API-strategie-extensies/ext-uri.md

```diff
@@ -0,0 +1,58 @@
+
+<div class="rule" id="api-59">
+  <p class="rulelab"><strong>API-59</strong>: Use spinal-case for path segments</p>
+  <p>In case path segments include compound words, the individual words are written in lower case and may be separated with hyphens. For example, a resource representing a collection of organisatie codes, results in either a path segment <code>organisatie-codes</code> or <code>organisatiecodes</code>.</p>
+</div>
+
+Also, the use of special characters used in path segments should be reduced to a minimum. Therefore, the use of diacritics and punctuation marks (other than hyphens, which are used in the spinal-case convention) must be avoided.
+
+<div class="rule" id="api-60">
+  <p class="rulelab"><strong>API-60</strong>: Normalize characters with diacritics to their base characters for path segments</p>
+  <p>In case path segments include diacritics (e.g. accents), these characters are normalized to their base characters. For example, a resource <code>scènes</code> results in path segment <code>scenes</code>.</p>
+</div>
+
+<div class="rule" id="api-67">
+  <p class="rulelab"><strong>API-67</strong>: Omit symbols and punctuation marks other than hyphens from path segments</p>
+  <p>Symbols and punctuation marks, other than hypens, must be omitted from path segments. For example:
+  <ul>
+    <li><code>https://api.example.org/v1/schemas</code> (<code>'</code> of schema's is omitted)</li>
+    <li><code>https://api.example.org/v1/hotel-restaurants</code> (<code>-</code> is preserved)</li>
+  </ul>
+  </p>
+</div>
+
+With regards to naming, resources must use clear, descriptive names that expresses what the respective resources represent. Therefore, the use of nonstandard abbreviations and superfluous prefixes should be avoided in resource names.
+
+<div class="rule" id="api-61">
+  <p class="rulelab"><strong>API-61</strong>: Do not explicitly indicate that a resource is an API</p>
+  <p>Use resource names as path segments (e.g. <code>gebouwen</code>) and do not explicitly mention that the resource is an API in the name (e.g. <code>gebouwen_api</code>).</p>
+</div>
+
+<div class="rule" id="api-62">
+  <p class="rulelab"><strong>API-62</strong>: Do not use file extensions in path segments</p>
+  <p>Preferably use content negotiation using the <code>Accept</code> header</p>
+</div>
+
+<div class="rule" id="api-63">
+  <p class="rulelab"><strong>API-63</strong>: Do not use nonstandard abbreviations as resource names</p>
+  <p>Avoid using nonstandard abbreviations for naming resources (e.g. <code>/coords</code> instead of <code>/coordinates</code>). This does not apply to standard or broadly used abbreviations (e.g. <code>bagNummeraanduiding</code>).</p>
+</div>
+
+### Query parameters
```

Comment:
Both recommendations clash directly with the DSO specification and HaalCentraal API's. 

Also in contrast with the use of spinal-case for path segments which "most API's" do use I don't see a strong preference for spinal-case query parameters in practice when looking at API's in general.

Reply:
Thanks @mtrimpe, which recommendations do you refer to? It does not become clear from the cited text above.

Reply:
Both recommendatins in the Query parameters section: spinal-case for query params conflict with DSO recommendation to use camelCasing and HaalCentraal which does the samen. 

DSO also specifically recommends using underscores as prefixes for variables which affect the format/behaviour of the API; similarly how the underscore is used/understood in the HAL standard.

Reply:
@frankvanes in mijn herinnering hebben we in de werkgroep gesproken over:
* spinal case voor path segments
* lower camel case voor query parameters
Ofwel gelijk aan het voorstel in https://github.com/Geonovum/KP-APIs/issues/273. Lower camel case voor query parameters zorgt ervoor dat je 1 op 1 JSON property namen kan gebruiken als query parameters. Het is ook beter interoperabel met client-side languages, zoals JavaScript.